org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/NameEnvironment.java

```diff
@@ -203,13 +216,25 @@ private void computeClasspathLocations(
 					}
 					bLocation.patchModuleName = patchedModuleName;
 				} else {
+					String release = Arrays.stream(entry.getExtraAttributes())
+							.filter(a -> IClasspathAttribute.RELEASE.equals(a.getName()))
+							.map(IClasspathAttribute::getValue).findFirst().orElse(null);
+					IContainer finalOutputFolder;
+					if (release == null) {
+						finalOutputFolder = outputFolder;
+					} else {
+						mustSortOutput = true;
+						finalOutputFolder = outputFolder.getFolder(new Path(String.format("META-INF/versions/%s", release))); //$NON-NLS-1$
+					}
+					int sourceTargetRelease = release==null?JavaProject.NO_RELEASE:Integer.parseInt(release);
```

Comment:
I can't help, but `sourceTargetRelease` sounds contradictory to me.

Reply:
renamed